lib/Dialect/HW/Transforms/HWSpecialize.cpp

```diff
@@ -243,9 +277,48 @@ static LogicalResult specializeModule(OpBuilder builder, ArrayAttr parameters,
       mapper.set(oldRes, newRes);
   }
 
+  // Register any nested parametric instance ops for the next pass
```

Comment:
This is now a fairly large function (~100loc) - could this lambda be meaningfully extracted to a separate function?

Reply:
It's now extracted it to a function